[BUG] CMake installation failed

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->

While packaging [Sysdig](https://github.com/draios/sysdig) v0.31.3 (a dependant of falcosecurity/libs) for Homebrew, CMake installation failed with the following error message (full logs available [here](https://github.com/Homebrew/homebrew-core/actions/runs/4467542973/jobs/7847277749#step:7:501)):

```plaintext
  -- Installing: /sysdig/userspace/libscap
  CMake Error at build/cmake_install.cmake:75 (file):
    file INSTALL cannot make directory "/sysdig/userspace/libscap": No such
    file or directory.
```

This is related to `libscap`'s CMake module [`libscap.cmake`](https://github.com/falcosecurity/libs/blob/003b600b551bb3c2546e363cc1873ed52a1bd7f2/cmake/modules/libscap.cmake). The core issue is that `libscap.cmake` uses `CMAKE_INSTALL_LIBDIR`, but the CMake module `GNUInstallDirs` is not `include`d.

Sysdig uses `libscap` [here](https://github.com/draios/sysdig/blob/eda4c141584a3d65f059f047312d6ee3c490fac2/cmake/modules/falcosecurity-libs.cmake#L99) and [here](https://github.com/draios/sysdig/blob/eda4c141584a3d65f059f047312d6ee3c490fac2/userspace/sysdig/CMakeLists.txt#L23). The `GNUInstallDirs` module is not included in either file.

**How to reproduce it**

<!-- Minimal and precise steps to reproduce the bug. -->

Sysdig 0.31.3 is built and installed with the following `cmake` invocations. falcosecurity/libs 0.10.5 is placed in the `falconsecurity-libs` directory, at the top level of Sysdig's source tree (`/private/tmp/sysdig-20230320-7324-6y7lgi/sysdig-0.31.3` in the following).

```
cmake -S . -B build \
  -DSYSDIG_VERSION=0.31.3 \
  -DUSE_BUNDLED_DEPS=OFF \
  -DCREATE_TEST_TARGETS=OFF \
  -DBUILD_LIBSCAP_EXAMPLES=OFF \
  -DDIR_ETC=/opt/homebrew/etc \
  -DFALCOSECURITY_LIBS_SOURCE_DIR=/private/tmp/sysdig-20230320-7324-6y7lgi/sysdig-0.31.3/falcosecurity-libs \
  -DCMAKE_CXX_STANDARD=17 \
  -DUSE_BUNDLED_CARES=OFF \
  -DUSE_BUNDLED_JSONCPP=OFF \
  -DUSE_BUNDLED_LUAJIT=OFF \
  -DUSE_BUNDLED_OPENSSL=OFF \
  -DUSE_BUNDLED_RE2=OFF \
  -DUSE_BUNDLED_TBB=OFF \
  -DUSE_BUNDLED_VALIJSON=OFF \
  -DUSE_BUNDLED_CURL=OFF \
  -DUSE_BUNDLED_NCURSES=OFF \
  -DUSE_BUNDLED_ZLIB=OFF \
  -DUSE_BUNDLED_B64=OFF \
  -DUSE_BUNDLED_GRPC=OFF \
  -DUSE_BUNDLED_JQ=OFF \
  -DUSE_BUNDLED_PROTOBUF=OFF \
  -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/sysdig/0.31.3 \
  -DCMAKE_INSTALL_LIBDIR=lib \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_FIND_FRAMEWORK=LAST \
  -DCMAKE_VERBOSE_MAKEFILE=ON \
  -Wno-dev \
  -DBUILD_TESTING=OFF \
  -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk
cmake --build build
cmake --install build
```

Both Sysdig and falcosecurity/libs sources were extracted from GitHub tarballs.

**Expected behaviour**

<!-- A clear and concise description of what you expected to happen. -->

Sysdig and falcosecurity/libs should be successfully installed.

**Screenshots**

<!-- If applicable, add screenshots to help explain your problem. -->

Not applicable, but full logs are available [here](https://github.com/Homebrew/homebrew-core/actions/runs/4467542973/jobs/7847277749#step:7:501). The error occurred [here](https://github.com/Homebrew/homebrew-core/actions/runs/4467542973/jobs/7847277749#step:7:390).

**Environment**

<!-- Please complete the following info. -->

- Falco version: The `falco` command was not installed; falcosecurity/libs (this repo) 0.10.5 was used.
<!-- Use "falco --version". -->
- System info: For macOS, full system information available [here](https://github.com/Homebrew/homebrew-core/actions/runs/4467542973/jobs/7847277749#step:6:22).
<!-- Falco has a built-in support command you can use  "falco --support | jq .system_info" -->
- Cloud provider or hardware configuration: Not on cloud; the build failed on macOS on both arm64 and x86_64, as well as on Ubuntu x86_64.
- OS: macOS 13.2 / Ubuntu 22.04
<!-- Eg., output of "cat /etc/os-release". -->
- Kernel: `Darwin Kernel Version 22.3.0` / `5.15.0-67-generic`
<!-- Eg., output of "uname -a". -->
- Installation method: From source; see above.
<!-- Eg., Kubernetes, RPM, DEB, from source? -->

**Additional context**

<!-- Add any other context about the problem here. -->

The error was observed in https://github.com/Homebrew/homebrew-core/pull/125925. This is currently worked around with the following patch:

```diff
--- a/cmake/modules/libscap.cmake
+++ b/cmake/modules/libscap.cmake
@@ -49,6 +49,9 @@ if(BUILD_LIBSCAP_MODERN_BPF)
        "${PROJECT_BINARY_DIR}/libpman/libpman.a"
 endif()
 )
+
+include(GNUInstallDirs)
+
 install(FILES ${LIBSCAP_LIBS} DESTINATION "${CMAKE_INSTALL_LIBDIR}/${LIBS_PACKAGE_NAME}"
                        COMPONENT "scap" OPTIONAL)
 install(DIRECTORY "${LIBSCAP_INCLUDE_DIR}" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${LIBS_PACKAGE_NAME}/userspace"
```

But this may be applicable to other modules in falcosecurity/libs, too. I believe that there are more missing CMake modules (e.g. `ExternalProject` missing in [`curl.cmake`](https://github.com/falcosecurity/libs/blob/003b600b551bb3c2546e363cc1873ed52a1bd7f2/cmake/modules/curl.cmake#L35)), but they may not be noticed due to transitive includes.

--- a/cmake/modules/libscap.cmake
+++ b/cmake/modules/libscap.cmake
@@ -8,6 +8,8 @@ endif()
 
 option(USE_BUNDLED_DEPS "Enable bundled dependencies instead of using the system ones" ON)
 
+include(GNUInstallDirs)
+
 include(ExternalProject)
 
 include(CheckSymbolExists)

